pallets/auctions/src/tests.rs

```diff
@@ -0,0 +1,788 @@
+
+            // should fail since the caller is not permitted to list on marketplace
+            let restricted_mkp_id =
+                help::create_mkp(bob.clone(), MarketplaceType::Private, 0, vec![1], vec![]);
+            assert_noop!(
+                Auctions::create_auction(
+                    alice.clone(),
+                    nft_id,
+                    restricted_mkp_id,
+                    ideal_start_block,
+                    ideal_end_block,
+                    100,
+                    Some(150)
+                ),
+                MarketplaceError::<Test>::NotAllowedToList
+            );
+        })
+}
+
+#[test]
+fn cancel_auction_happy() {
+    ExtBuilder::default()
+        .caps(vec![(ALICE, 1000), (BOB, 1000)])
+        .build()
+        .execute_with(|| {
+            let alice: mock::Origin = RawOrigin::Signed(ALICE).into();
+
+            let mkp_id = get_marketplace(ALICE);
+            let nft_id = create_nft(ALICE);
+            create_auction(alice.clone(), mkp_id, nft_id);
+
+            assert_ok!(Auctions::cancel_auction(alice, nft_id,));
+        })
+}
+
+#[test]
+fn cancel_auction_unhappy() {
+    ExtBuilder::default()
+        .caps(vec![(ALICE, 1000), (BOB, 1000)])
+        .build()
+        .execute_with(|| {
+            let alice: mock::Origin = RawOrigin::Signed(ALICE).into();
+            let bob: mock::Origin = RawOrigin::Signed(BOB).into();
+
+            let mkp_id = get_marketplace(ALICE);
+            let nft_id = create_nft(ALICE);
+
+            // should fail since the auction does not exist
+            assert_noop!(
+                Auctions::cancel_auction(alice.clone(), nft_id,),
+                Error::<Test>::AuctionDoesNotExist
+            );
+
+            create_auction(alice.clone(), mkp_id, nft_id);
+
+            // should fail since the caller is not creator of auction
+            assert_noop!(
+                Auctions::cancel_auction(bob.clone(), nft_id,),
+                Error::<Test>::OnlyAuctionCreatorCanCancel
+            );
+
+            // should fail since the auction has already started
+            run_to_block(MIN_AUCTION_BUFFER + 2);
+            assert_noop!(
+                Auctions::cancel_auction(alice, nft_id,),
+                Error::<Test>::CannotCancelInProcessAuction
+            );
+        })
+}
+
+#[test]
+fn add_bid_happy() {
+    ExtBuilder::default()
+        .caps(vec![(ALICE, 1000), (BOB, 1000), (CHARLIE, 1000)])
+        .build()
+        .execute_with(|| {
+            let alice: mock::Origin = RawOrigin::Signed(ALICE).into();
+            let bob: mock::Origin = RawOrigin::Signed(BOB).into();
+            let charlie: mock::Origin = RawOrigin::Signed(CHARLIE).into();
+
+            let mkp_id = get_marketplace(ALICE);
+            let nft_id = create_nft(ALICE);
+            create_auction(alice, mkp_id, nft_id);
+
+            run_to_block(MIN_AUCTION_BUFFER + 2);
+            assert_ok!(Auctions::add_bid(bob, nft_id, 102));
```

Comment:
You need to assert the new auction pallet internal state .
You need to assert that the bidder has now less tokens in his balance.